### DISCUSSION TITLE
Memory full

### DISCUSSION THREAD

**Discussion by Rhoenit** [telekom-security/tpotce#886] [Q&A]
I´m trying to use tpot in a VM but after 1-2 weeks the memory is full.  
Does anyone have an idea whats wrong or how to clean up?

df-h 
![df-h](https://user-images.githubusercontent.com/78480832/131662376-53fff583-03fd-4951-a4d6-67ffeb01ad27.PNG)

logstash
![logstash1](https://user-images.githubusercontent.com/78480832/131662411-7ce73f82-ff5f-4c36-b828-99c9b14f0f9c.PNG)
![logstash2](https://user-images.githubusercontent.com/78480832/131662417-66b7daba-df87-4257-b5bd-af96f2c90dbd.PNG)
![logstash3](https://user-images.githubusercontent.com/78480832/131662420-954786a2-c140-4712-b05c-43ce8c71fde2.PNG)

**Comment by t3chn0m4g3** [comment]
In `/opt/tpot/bin` you may find the `clean.sh` which can delete all logs from the honeypots.

In `/etc/crontab` look for the following lines and adjust the configs to your needs.
```
# Delete elasticsearch logstash indices older than 90 days
$myRANDOM_MINUTE $myDEL_HOUR * * *      root    curator --config /opt/tpot/etc/curator/curator.yml /opt/tpot/etc/curator/actions.yml
```

**Reply by Rhoenit** [reply]
after executing clean.sh it still has so much memory used. I really don´t know what takes so much space. It seems like it should be everything in /var/lib/docker/overlay2/.... 

even after deleting every elasticsearch logstash indices it is still full.
Do you have any other idea?

**Reply by t3chn0m4g3** [reply]
Orphaned docker images? `# docker images`
It is also possible that some system logs are not rotated correctly, you have to look for such folders with `# du -a / | sort -n -r | head -n 20` (adjust this to your needs).

**Reply by Rhoenit** [reply]
does this help?

![du](https://user-images.githubusercontent.com/78480832/132000741-9b42a9de-3fef-42f2-8c13-8691016e0836.PNG)

**Reply by t3chn0m4g3** [reply]
Does not seem to be too much of data in the logs.

**Reply by Rhoenit** [reply]
this tpot was fresh installed 3 days ago and already uses 89GB. The biggest logstash is 22mb. 
Can you explain this and how to prevent?

**Reply by t3chn0m4g3** [reply]
Given the fact that as per your feedback the docker images, honeypot logs, elk data and system logs are not the culprit it is highly atypical and should look like this (install is running for about two week with very high volume of events):
![image](https://user-images.githubusercontent.com/4318452/132005884-4bb925d0-1d52-4857-99a6-06f8e2a1ba05.png)

**Reply by Rhoenit** [reply]
Now it is running for a week and I guess it is running out of space again this week. I already used clean.sh. 
There aren´t a lot of events. Only 4 CiscoASA. 
What is using so much space? 

![df-hv](https://user-images.githubusercontent.com/78480832/132313058-51d2244a-8e8d-4b7b-9b43-3ba010b28677.PNG)
![kibana30days](https://user-images.githubusercontent.com/78480832/132313069-71aab0ab-dfc7-4465-abd5-c28348beed2f.PNG)

**Comment by t3chn0m4g3** [comment]
I cannot reproduce this. However you can try [this ](https://stackoverflow.com/questions/61097403/bash-treesize-script-improved-output-alignment#61098098) and iteratively follow the folders using the most disk space. Once the folders and files are identified we have a clue to continue the investigation.

**Comment by Rhoenit** [comment]
I found this: 

![harddrive](https://user-images.githubusercontent.com/78480832/132322129-0f9cd599-12c2-4002-b86a-c10807104160.PNG)
![varsize](https://user-images.githubusercontent.com/78480832/132322180-8dda5a6d-b2fb-42b0-9176-cc6b9541bc40.PNG)
![size](https://user-images.githubusercontent.com/78480832/132322209-338bd640-b50d-4d9b-b3be-400be41dedcf.PNG)

**Reply by t3chn0m4g3** [reply]
For some reason your `daemon.log` is unusually large. Please `tail` it for contents. The rest looks pretty much fine, even though `syslog` is quite large.

**Comment by Rhoenit** [comment]
![tail](https://user-images.githubusercontent.com/78480832/132324310-5f6e65ef-4edf-4fdb-b586-cb47779f31ae.PNG)

**Comment by t3chn0m4g3** [comment]
Is it possible that for some reason the machine is constantly restarting the `tpot.service` (`dps.sh -2`). Based on the log messages it seems that some services are constantly restarting. At least this would explain the high volume of logs.

This installation is running since December of 2020, without any manual cleaning interaction and this is how it should look like:
![image](https://user-images.githubusercontent.com/4318452/132326249-560dc0aa-17df-438d-be38-17c5edb65ac8.png)

**Reply by Rhoenit** [reply]
it only looks like ewsposter restarting all the time. 
![dpssh](https://user-images.githubusercontent.com/78480832/132332942-ddd7a9e0-b10f-452e-9ae2-635fa77580c7.PNG)

**Comment by t3chn0m4g3** [comment]
That is a known [issue](https://github.com/telekom-security/ewsposter/issues/4) and @trixam is working on a fix.

**Comment by t3chn0m4g3** [comment]
Meanwhile you can adjust `/etc/logrotate.d/rsyslog` to your [needs ](https://linux.die.net/man/8/logrotate) so that `deamon.log` will either be rotated earlier or deleted.
